File: C++/Tree/Tree_01.cpp
/*
===============================================================================
Program Name    : Binary Search Tree Skeleton
Program Number  : 001
File Name       : Tree_01.cpp

Description     : This program demonstrates the basic
                  structure of Binary Search Tree (BST).

                  It contains:
                  1. Node Structure
                  2. BST Class
                  3. Constructor

Concepts Used:
1. Self Referential Structure
2. Class and Object
3. Dynamic Memory Management
4. Binary Search Tree

Date            : August 2026
Language        : C++
Author          : Atharv Tushar Bhosale
===============================================================================
*/

// BST SKELETON

#include<iostream>

using namespace std;

#pragma pack(1)

struct node
{
    int data;
    struct node *lchild;
    struct node *rchild;
};

class BST
{
    private:
        struct node *first;

    public:
        BST();
};

///////////////////////////////////////////////////////////////////////////////
//
// Constructor
//
///////////////////////////////////////////////////////////////////////////////

BST :: BST()
{
    first = NULL;
}

///////////////////////////////////////////////////////////////////////////////
//
// Entry Point Function
//
///////////////////////////////////////////////////////////////////////////////

int main()
{
    BST bobj;

    cout<<"Binary Search Tree Skeleton Created Successfully\n";

    return 0;
}

/*
===============================================================================
Sample Output

Binary Search Tree Skeleton Created Successfully

===============================================================================
End of Program
===============================================================================
*/

File: Java/Tree/Tree_01.java
/*
===============================================================================
Program Name    : Binary Search Tree Skeleton
Program Number  : 001
File Name       : Tree_01.java

Description     : This program demonstrates the basic
                  structure of a Binary Search Tree
                  using self-referential nodes.

Concepts Used:
1. Class and Object
2. Self-Referential Class
3. Binary Search Tree
4. Left and Right Child References

Date            : August 2026
Language        : Java
Author          : Atharv Tushar Bhosale
===============================================================================
*/

class node
{
    public int data;
    public node lchild;
    public node rchild;
}

class Tree
{
    private node first;

    public Tree()
    {
        first = null;
    }
}

class Tree_01
{
    public static void main(String A[])
    {
        Tree tobj = new Tree();

        System.out.println("Binary Search Tree created successfully");
    }
}

/*
===============================================================================
Sample Output

Binary Search Tree created successfully

===============================================================================
End of Program
===============================================================================
*/
